crates/release_plz_core/src/project.rs

```diff
@@ -171,21 +172,53 @@ impl Project {
             if package.description.is_none() {
                 missing_fields.push(format!("- `description` for package `{}`", package.name));
             }
-        }
 
-        if missing_fields.is_empty() {
-            Ok(())
-        } else {
-            let error_message = format!(
+            check_local_dependencies(package, &mut missing_versions);
+        }
+        let has_missing_fields = !missing_fields.is_empty();
+        let has_missing_version = !missing_versions.is_empty();
+        if !has_missing_fields && !has_missing_version {
+            return Ok(());
+        }
+        let mut error_message = String::new();
+        if has_missing_fields {
+            error_message.push_str(&format!(
                 "The following mandatory fields for crates.io are missing in Cargo.toml:
 {}
-See https://doc.rust-lang.org/cargo/reference/manifest.html
-
-Note: to disable this check, set the `--no-toml-check` flag.",
+See https://doc.rust-lang.org/cargo/reference/manifest.html\n",
                 missing_fields.join("\n")
-            );
-            anyhow::bail!(error_message);
+            ));
+        }
+        if has_missing_version {
+            error_message.push_str(&format!(
+                "The following packages have local dependencies missing a version specifier:
+{}",
+                missing_versions.join("\n")
+            ));
         }
+        error_message.push_str("\nNote: to disable this check, set the `--no-toml-check` flag.");
+        anyhow::bail!(error_message);
+    }
+}
+
+fn check_local_dependencies(package: &Package, missing_versions: &mut Vec<String>) {
```

Comment:
can you return `missing_versions` from this function instead of taking it as input?
I think it's more clear.

I use this pattern mainly for recursive functions or similar

Reply:
Done.